ui/app/src/Views/AlertList/Filter.elm

```diff
@@ -2,14 +2,26 @@ module Views.AlertList.Filter exposing (receiver, silenced, matchers)
 
 import Alerts.Types exposing (Alert, AlertGroup, Block)
 import Utils.Types exposing (Matchers)
-import Regex
+import Utils.Filter exposing (Matcher, MatchOperator(Eq, NotEq, RegexMatch, NotRegexMatch))
+import Regex exposing (regex, contains)
 
 
-receiver : Maybe String -> List AlertGroup -> List AlertGroup
+receiver : Maybe Matcher -> List AlertGroup -> List AlertGroup
 receiver maybeReceiver groups =
     case maybeReceiver of
-        Just receiver ->
-            by (filterAlertGroup receiver) groups
+        Just { key, op, value } ->
+            case op of
+                Eq ->
+                    by (filterAlertGroup ((==) value)) groups
+
+                RegexMatch ->
+                    by (filterAlertGroup ((regex >> contains) value)) groups
+
+                NotEq ->
+                    groups
```

Comment:
I made the comment in the description that `NotEq` and `NotRegexMatch`aren't supported, but didn't want to have a `_ ->` match group to catch everything.